File: 5Days-SoftwareArchitecture/typesript/Day2/Connascence/con_type.ts
class PaymentGateway {
    transfer(amount: number) {
        console.log(`Transferred ₹${amount}`);
    }
}

class ClientType {
    private gateway = new PaymentGateway();

    makeTransfer() {
        this.gateway.transfer(2500.50); // must match type: number
    }
}

new ClientType().makeTransfer();

File: 5Days-SoftwareArchitecture/Java/Day 2/Connascence/CoN_Type.java
class PaymentGateway {
    void transfer(double amount) { // expects double
        System.out.println("Transferred ₹" + amount);
    }
}

class ClientType {
    private PaymentGateway gateway = new PaymentGateway();

    void makeTransfer() {
        gateway.transfer(2500.50); // must pass correct type: double
    }
}

public class CoN_Type {
    public static void main(String[] args) {
        new ClientType().makeTransfer();
    }
}
